### PR TITLE
[datadog_team] Add datadog team data sources

### PR DESCRIPTION
**Pull request by skarimo**
This PR adds the following 2 data sources:

- datadog_team
- datadog_team_memberships

This is a follow up to: https://github.com/DataDog/terraform-provider-datadog/pull/1908

### CODE REVIEW

File: datadog/fwprovider/data_source_datadog_team_memberships.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+		optionalParams.PageSize = &pageSize
+
+		ddResp, _, err := d.Api.GetTeamMemberships(d.Auth, teamID, optionalParams)
+		if err != nil {
+			resp.Diagnostics.Append(utils.FrameworkErrorDiag(err, "error getting team memberships"))
+			return
+		}
+
+		userTeams = append(userTeams, ddResp.GetData()...)
+		if len(ddResp.GetData()) == 0 || len(ddResp.GetData()) < 100 {
```
Comment (nkzou) [inline datadog/fwprovider/data_source_datadog_team_memberships.go:119]:
```suggestion
		if len(ddResp.GetData()) < 100 {
```

File: datadog/fwprovider/data_source_datadog_team.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+	r.Auth = providerData.Auth
+}
+
+func (d *DatadogTeamDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "team"
+}
+
+func (d *DatadogTeamDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Use this data source to retrieve information about an existing api key.",
```
Comment (nkzou) [inline datadog/fwprovider/data_source_datadog_team.go:62]:
```suggestion
		Description: "Use this data source to retrieve information about an existing Datadog team.",
```

File: datadog/fwprovider/data_source_datadog_team_memberships.go [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+	r.Auth = providerData.Auth
+}
+
+func (d *DatadogTeamMembershipsDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "team_memberships"
+}
+
+func (d *DatadogTeamMembershipsDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Use this data source to retrieve information about an existing api key.",
```
Comment (nkzou) [inline datadog/fwprovider/data_source_datadog_team_memberships.go:59]:
```suggestion
		Description: "Use this data source to retrieve information about an existing Datadog team membership.",
```

File: datadog/fwprovider/data_source_datadog_team_memberships.go [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+				Required:    true,
+			},
+			"filter_keyword": schema.StringAttribute{
+				Description: "Search query, can be user email or name.",
+				Optional:    true,
+			},
+			// Computed values
+			"team_memberships": schema.ListAttribute{
+				Computed:    true,
+				Description: "Free-form markdown description/content for the team's homepage.",
```
Comment (nkzou) [inline datadog/fwprovider/data_source_datadog_team_memberships.go:75]: Is this description right? Doesn't seem to match

**Review by nkzou — CHANGES_REQUESTED** [review]

File: datadog/fwprovider/data_source_datadog_team.go [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+				Optional:    true,
+			},
+			// Computed values
+			"description": schema.StringAttribute{
+				Computed:    true,
+				Description: "Free-form markdown description/content for the team's homepage.",
+			},
+			"handle": schema.StringAttribute{
+				Computed:    true,
+				Description: "The team's identifier",
```
Comment (KrisKennawayDD) [inline datadog/fwprovider/data_source_datadog_team.go:83]: nit: s/identifier/handle/

**Review by KrisKennawayDD — APPROVED** [review]

**Review by eric-wood — APPROVED** [review]
This looks good to me!

**Pull request merged**